### PR TITLE
Update @hatchify/react docs page

### PR DESCRIPTION
**Pull request by aminnoah**
# Description

# Jira ticket links

# Test Plan

# Definition of done

- [x] In depth review

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,11 +220,11 @@ Continue learning more about the Hatchify feature set with these guides that con
 
 Learn how to make Hatchify match your needs with its technical interface documentation:
 
-- [Schema](docs/schema/)
+- [Schema](docs/schema/) - Learn how to specify attributes and relationships with Hatchify's interface for defining your data, called "schemas"
   - [Attributes](docs/schema/attribute-types/README.md)
   - [Relationships](docs/schema/relationship-types/README.md)
   - [Schema Naming](docs/schema/naming.md)
-- [JSON:API](docs/jsonapi/README.md)
+- [JSON:API](docs/jsonapi/README.md) - Learn the details of our JSONAPI implementation.
```
Comment (arthurpankiewicz) [inline README.md:227]: should be `JSON:API` implementation rather than `JSONAPI`

File: README.md [hunk 2]
```diff
@@ -234,14 +234,14 @@ Learn how to make Hatchify match your needs with its technical interface documen
...
   - [Updating](docs/jsonapi/updating.md)
   - [Deleting](docs/jsonapi/deleting.md)
-- Koa
+- Koa - Learn how Hatchify leverages Koa middleware functionality
   - [API Docs](docs/koa/README.md)
-- Express
+- Express - Learn the details of our Express implementation
   - [API Docs](docs/express/README.md) 🛑
-- React
+- React - Learn how to use and customize Hatchify's schema-driven library of React components and helper methods
```
Comment (arthurpankiewicz) [inline README.md:241]: `helper methods` -> `CRUD methods` or `CRUD functions`

File: docs/react/README.md [hunk 3]
```diff
@@ -1,72 +1,186 @@
 # @hatchifyjs/react
 
-- [What is @hatchifyjs/react](#what-is-hatchifyjsreact)
-  - [TypeScript](#typescript)
-  - [Types](#types)
+@hatchifyjs/react is a schema-driven library of components and helper methods for a Hatchify + React frontend. By providing the schemas (AKA models) of your backend resources, @hatchifyjs/react returns a set of components and helper functionality that you can use across your React app.
```
Comment (arthurpankiewicz) [inline docs/react/README.md:3]: helper functionality can just be `CRUD functions and hooks`

File: docs/react/README.md [hunk 4]
```diff
@@ -1,72 +1,186 @@
...
+
+`@hatchifyjs/react` is a TypeScript supported [NPM package](https://www.npmjs.com/package/@hatchifyjs/react) that takes your [JSON:API rest client](#createjsonapiclient) and[Schema(s)](../schema/README.md) to produce custom:
```
Comment (arthurpankiewicz) [inline docs/react/README.md:5]: i'd word it along the lines of:

`@hatchifyjs/react` is an [NPM package](https://www.npmjs.com/package/@hatchifyjs/react) that takes your schemas and provides:

- Components
- CRUD promise functions and react hooks
- TypeScript

File: docs/react/README.md [hunk 5]
```diff
@@ -1,72 +1,186 @@
...
+- components
+- helper functions
+- generated [types](#types)
+
+The following uses `hatchifyReact`to create a `hatchedReact` app with the defined `Todo` and `User` schemas.
+
+```tsx
+import { hatchifyReact, HatchifyProvider, createJsonapiClient } from "@hatchifyjs/react"
+
+// Define your Schema(s)
```
Comment (arthurpankiewicz) [inline docs/react/README.md:16]: i think we can remove this comment, and make `Schemas` lowercase `s`

File: docs/react/README.md [hunk 6]
```diff
@@ -1,72 +1,186 @@
...
+      todos: hasMany(),
+    },
+  },
+} satisfies Record<string, PartialSchema>
+
+// Create your Hatched React App instance
+const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
+
+// Define variables for your Hatchify schema generated components
+const TodoDataGrid = hatchedReact.components.Todo.DataGrid
```
Comment (arthurpankiewicz) [inline docs/react/README.md:43]: // Define variables for your Hatchify schema generated components

can just be

// Grab the Todo DataGrid component from the hatchedReact object

File: docs/react/README.md [hunk 7]
```diff
@@ -1,72 +1,186 @@
...
+
+// Render your React Functional Component
+const HatchedComponent = () => {
```
Comment (arthurpankiewicz) [inline docs/react/README.md:46]: can remove the comment, and just make this App, since that's what's being exported on line 57:

`function App() {`

File: docs/react/README.md [hunk 8]
```diff
@@ -1,72 +1,186 @@
...
+  return (
+    <HatchifyProvider>
+      <button onClick={onActionClick} style={{ margin: 10 }}>
```
Comment (arthurpankiewicz) [inline docs/react/README.md:49]: remove this button, since we don't have the selected state/props setup

File: docs/react/README.md [hunk 9]
```diff
@@ -1,72 +1,186 @@
...
+        action
+      </button>
+      <TodoDataGrid defaultSort={{ direction: "asc", sortBy: "id" }}></TodoDataGrid>
```
Comment (arthurpankiewicz) [inline docs/react/README.md:52]: i think for simplicity on this initial example, we just do `<TodoDataGrid />`

File: docs/react/README.md [hunk 10]
```diff
@@ -1,72 +1,186 @@
...
+    </HatchifyProvider>
+  )
+}
+
+export default App
+```
+
+[!IMPORTANT]
+The `HatchifyProvider` wraps all Hatchify components to manage their internal states. See [HatchifyProvider](#hatchifyprovider) for more information.
```
Comment (arthurpankiewicz) [inline docs/react/README.md:61]: `HatchifyProvider` doesn't manage state, it's for figuring out which component (design system to use (user doesn't really need to know this)) and the default components for different types. i'd word it maybe like this:

You must wrap your `App` in a `HatchifyProvider` so that you may use the components provided by Hatchify.

Comment (arthurpankiewicz) [reply docs/react/README.md:61]: also i just checked, the `HatchifyProvider` that is imported from `@hatchifyjs/react` doesn't provide a way to override defaults lol - only the `HatchifyPresentationProvider` which is not exported, i will make a backlog ticket for this

File: docs/react/README.md [hunk 11]
```diff
@@ -1,72 +1,186 @@
...
-      - [useDataGridState](#usedatagridstate)
-  - [MUI Components](#mui-components)
+  - [Types](#types)
+    - [DataGridState](#usedatagridstate)
+    - [CreateType](#usedatagridstate)
+    - [UpdateType](#usedatagridstate)
+    - [HatchifyApp](#usedatagridstate)
+    - [RecordType](#usedatagridstate)
+- [`hatchedReact`](#hatchedreact)
+  - [`hatchedReact.Everything`](#everything)
```
Comment (arthurpankiewicz) [inline docs/react/README.md:74]: since these are nested under `hatchedReact` i think you can remove the prefix, so we just have:

- hatchedReact
  - Everything
  - components
  ...

File: docs/react/README.md [hunk 12]
```diff
@@ -1,72 +1,186 @@
...
+  - [`hatchedReact.components`](#components)
+  - [`hatchedReact.model`](#model)
+  - [`hatchedReact.state`](#state)
+    - [useDataGridState](#usedatagridstate)
+- [MUI Components](#mui-components)
+
+- [Exports](#exports)
+  - createJsonapiClient - Creates a new [JSON:API rest client](#createjsonapiclient) using the defined schemas
+  - hatchifyReact - Constructs a `hatchedReact` app instance with custom components,helper functions, and type definitions
+  - HatchifyProvider - A component that hosts and provides access to Hatchify-related state
```
Comment (arthurpankiewicz) [inline docs/react/README.md:84]: these should be links and not definitions, since this is the table of contents

File: docs/react/README.md [hunk 13]
```diff
@@ -1,72 +1,186 @@
...
+- [`hatchedReact`](#hatchedreact)
```
Comment (arthurpankiewicz) [inline docs/react/README.md:85]: i think this section is duplicated from line 73

File: docs/react/README.md [hunk 14]
```diff
@@ -1,72 +1,186 @@
...
+Returns a `JSON:API rest client` instance object
 
-## createJsonapiClient
+### hatchifyReact
 
-`createJsonapiClient` creates a new JSON:API rest client from the defined schemas. It accepts a base url, and schema set. For more documentation see [here](./rest-client.md) 🛑.
+`hatchifyReact(createJsonapiClient("/api", Schemas))` is a `Function` that initializes the `HatchifyApp` object from the JSON:API rest client. Inside of the retured object you will find [`components`](./components.md), [`model`](./model.md), [`state`](./state.md), and [`Everything`](#everything).
 
 ```ts
+import {createJsonapiClient, hatchifyReact} from "@hatchifyjs/react"
```
Comment (arthurpankiewicz) [inline docs/react/README.md:123]: inconsistent styling, there should be a space after/before the `{...}`

File: docs/react/README.md [hunk 15]
```diff
@@ -1,72 +1,186 @@
...
+const schemas = { ... }
+
 const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
 ```
 
-## HatchifyProvider
+### HatchifyProvider
 
-`HatchifyProvider` is a component that hosts and provides access to Hatchify-related state. Hatchify components in `HatchifyProvider`'s subtree interact with it to access Hatchify internal state. It must be a parent to any Hatchify components.
+`HatchifyProvider` is a component that hosts and provides access to Hatchify-related state. It must be a parent to any Hatchify components.
```
Comment (arthurpankiewicz) [inline docs/react/README.md:132]: i'd reword this similarly to what i commented above

File: docs/react/README.md [hunk 16]
```diff
@@ -1,72 +1,186 @@
...
+
+const schemas = { ... }
+
+// Create the Hatched React App instance
+const hatchedReact = hatchifyReact(createJsonapiClient("/api", Schemas))
+
+// Define variables for your Hatchify schema generated components
+const TodoDataGrid = hatchedReact.components.Todo.DataGrid
+
+const HatchedComponent = () => {
```
Comment (arthurpankiewicz) [inline docs/react/README.md:147]: same stuff in this snippet and others - should just be `function App()` everywhere, it needs the mui imports for provider and wrap it.

Comment (arthurpankiewicz) [reply docs/react/README.md:147]: this code snippet is almost the same as the one at the top of the page - maybe we don't need it?

File: docs/react/README.md [hunk 17]
```diff
@@ -1,72 +1,186 @@
...
 }
 ```
 
-## hatchifyReact
+### Types
+
+`@hatchifyjs/react` provides a number of [types](./types.md) to assist with app customization.
 
-`hatchifyReact` is a function that initializes the `HatchifyApp` object. Inside of the object you will find `components`, `model`, `state`, and `Everything`. `components`, `model`, and `state`, will be broken down further for each defined schema
+| key                                         | description                                                                             |
```
Comment (arthurpankiewicz) [inline docs/react/README.md:160]: instead of `key` make it `Type`

File: docs/react/README.md [hunk 18]
```diff
@@ -1,72 +1,186 @@
...
+| ------------------------------------------- | --------------------------------------------------------------------------------------- |
+| [`DataGridState`](./types.md#datagridstate) | The return type of the `useDataGridState` hook.                                         |
+| [`CreateType`](./types.md#createtype)       | The type used when data is created.                                                     |
+| [`UpdateType`](./types.md#updatetype)       | Is the type used when data is updated.                                                  |
+| [`HatchifyApp`](./types.md#hatchifyapp)     | the type of the Hatchify App that is generated with [`hatchifyReact()`](#hatchifyreact) |
+| [`RecordType`](./types.md#recordtype)       | Matches a record's schema                                                               |
```
Comment (arthurpankiewicz) [inline docs/react/README.md:166]: for the details bit i'd say:

DataGridState - The return type of `DataGridState`. Can be used if you are extending the functionality of `useDataGridState`.

CreateType - This can be used to type data that will be used to create a record for a given schema

UpdateType - ^ same as above but "update" instead of "create"

HatchifyApp - The return type of the object that is generated by `hatchifyReact`

RecordType - The type of a record for a given schema

File: docs/react/README.md [hunk 19]
```diff
@@ -1,72 +1,186 @@
...
+Learn more about the available types [here](./types.md).
+
+## `hatchedReact`
+
+`hatchedReact` is the instantiated [`HatchifyReact` App](#hatchifyreact) that is returned by the [`hatchifyReact`] constructor function. It provides:
+
+- [Everything](./Everything.md) - is a default component comprised of a set of tabs 🛑 (one for each schema) and DataGrids.
+- [components](./components.md) - a set of components for each of the defined schemas to be used in the app.
+- [model](./model.md) - a set of hooks and promises for each of the defined schemas
+- [state](./state.md) - a set of hooks for each of the defined schemas to be used in the app.
```
Comment (arthurpankiewicz) [inline docs/react/README.md:177]: state - contains hooks that can be used when customizing components provided by hatchify

File: docs/react/README.md [hunk 20]
```diff
@@ -90,9 +204,39 @@ Included components:
 
 Learn more about the available components [here](./components.md).
```
Comment (arthurpankiewicz) [inline docs/react/README.md:206]: we should define the `DataGrid`, `Column`, and `Empty` components and whatever props they take, or if these are defined elsewhere already, then we should link to them directly

File: docs/react/README.md [hunk 21]
```diff
@@ -90,9 +204,39 @@ Included components:
...
+
+// Render your custom component 
+const HatchedComponent = () => {
+  return (
+    <HatchifyProvider>
+      <TodoEmpty />
+      <TodoDataGrid>
+        <TodoColumn />
+      </TodoDataGrid>
+      <TodoEmpty />
```
Comment (arthurpankiewicz) [inline docs/react/README.md:231]: `TodoEmpty` is a compound component used for customizing a `TodoDataGrid` - it's what you want to render if the list is empty (no records found. `TodoColumn` is used to customize an existing column or add one, so i'd do something like:

```
<TodoDataGrid>
  <TodoEmpty>No todos found!</TodoEmpty>
  <TodoColumn
    label="Actions"
    renderDataValue={() => <button onClick={({ record }) => console.log(record)}>Log Todo</button>
  />
</TodoDataGrid>
```

File: README.md [hunk 22]
```diff
@@ -234,14 +234,14 @@ Learn how to make Hatchify match your needs with its technical interface documen
     - [Sparse Fields](docs/jsonapi/reading/sparse-fields/README.md)
   - [Updating](docs/jsonapi/updating.md)
   - [Deleting](docs/jsonapi/deleting.md)
-- Koa
+- Koa - Learn how Hatchify leverages Koa middleware functionality
   - [API Docs](docs/koa/README.md)
-- Express
+- Express - Learn the details of our Express implementation
```
[the remark this answers is not in the document]
Comment (aminnoah) [reply README.md:239]: I might need you to make a decision on that one. What are your thoughts?

File: docs/react/README.md (hunk 10 above)
Comment (aminnoah) [reply docs/react/README.md:61]: Will we be exposing the `HatchifyPresentationProvider` or just allow the `HatchifyProvider` to override defaults
